app/profile/tests/test.form.modal/test.form.modal.controller.js

```diff
@@ -6,13 +6,39 @@
         .module('app')
         .controller('TestFormModalController', TestFormModalController);
 
-    function TestFormModalController ($uibModalInstance) {
+    function TestFormModalController ($uibModalInstance, $http, Endpoints, growl, $location, $log) {
         const vm = this;
 
         vm.close = close;
+        vm.submit = submit;
+
+        ////
 
         function close(data) {
             $uibModalInstance.close(data);
         }
+
+        function submit () {
+
+            if (!vm.form || vm.form.$invalid) {
+                return vm.form.$submitted = false;
+            }
+
+            let data = {
+                name: vm.test.name,
+                shortDescription: vm.test.shortDescription,
+                description: vm.test.description,
+                testType: vm.test.testType
+            };
+
+            $http.post(Endpoints.ACHIEVEMENTS_TEST, data).then((res) => {
```

Comment:
See, you definitely need to create a factory for ACHIEVEMENTS_TEST. It allows you to use one endpoint in many places.